src/components/MySongs/MySongs.js

```diff
@@ -109,15 +114,15 @@ export class MySongsComponent extends SearchFunctionalityProviderComponent {
   ) {
     if (shouldFetchData) {
       this.state.isFetching = true;
-      Promise.resolve(this.fetchSongs())
-        .then(() => this.mount(false))
-        .then(() => this.props.onDataReceived(this.songs));
+      Promise.resolve(this.fetchSongs()).then(() => this.mount(false));
```

Comment:
there is no need to wrap this into Promise.resolve()

just this.fetchSongs().then(() => this.mount(false)); is enough

Reply:
Done. Thank you